modules/planning/tasks/deciders/path_assessment_decider.cc

```diff
@@ -454,18 +455,19 @@ void PathAssessmentDecider::SetObstacleDistance(
 
   // Get all obstacles and convert them into frenet-frame polygons.
   std::vector<Polygon2d> obstacle_polygons;
-  auto indexed_obstacles = reference_line_info.path_decision().obstacles();
+  const auto& indexed_obstacles =
+      reference_line_info.path_decision().obstacles();
   for (const auto* obstacle : indexed_obstacles.Items()) {
     // Filter out unrelated obstacles.
     if (!IsWithinPathDeciderScopeObstacle(*obstacle)) {
       continue;
     }
     // Convert into polygon and save it.
-    const auto obstacle_box = obstacle->PerceptionBoundingBox();
+    const auto& obstacle_box = obstacle->PerceptionBoundingBox();
     if (obstacle_box.area() < kMinObstacleArea) {
       continue;
     }
-    obstacle_polygons.push_back(Polygon2d(obstacle_box));
+    obstacle_polygons.emplace_back(obstacle_box);
```

Comment:
There are two constructors taking different inputs for Polygon2d. Will "emplace_back" work properly here?